_rules/table-headers-attribute-refer-to-data-cells.md

```diff
@@ -0,0 +1,208 @@
+---
+name: `headers` attribute only refers to `cells` in the same `table` element
+
+test_type: atomic
+
+description: |
+  This rule checks that the `headers` attribute must only refer to `cells` in the same `table` element
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Test Procedure
+
+### Applicability
+
+This rule applies to `td` and `th` elements with a `headers` attribute and within a `table` element, that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+### Expectation
+
+Each target element with the `headers` attribute refers to other `cells` of the same `table` element.
```

Comment:
I don't think the table would fail WCAG if a header attribute refers to a non existing cell but there was no relationship to communicate anyway. What would be the harm in that WCAG wise?

Doesn't this rule need the same "the headers attribute is being used to comply to WCAG" assumption like all the other "spellchecking" rules?

Reply:
I think that's the [F90](https://www.w3.org/WAI/WCAG21/Techniques/failures/F90) : Failure of Success Criterion 1.3.1 for incorrectly associating table headers and content via the headers and id attributes .

"The failure occurs when the relationship between data cells and corresponding header cells cannot be programmatically determined correctly because **the association of id and headers attributes is faulty.**"

There's a real issue on accessing the information if the relationship between heading and cell is invalid, because users that rely only on AT (screen reader) may access to wrong informations, and they do not have any other way to access the correct information (eg the visual presentation of the table for example). Those users may never find or find a wrong information if those relationships are wrong, and they will never know the information is wrong. 

(And I think we should having that F90 reference in the background section?)